### ISSUE TITLE
Avoid Conflicts between "member/provider export" and Salesforce Integration.

### ISSUE THREAD

**Issue by kjgarza** [datacite/datacite#1271]
The sync creates an organization before the Salesforce integration can update the values obtained by the integration. Therefore a second organization is created with the account values and when the salesforce integration wants to modify them is not possible as one cannot have multiple fabrica IDs.

Details of the investigation: https://docs.google.com/document/d/11DZovE3HX0JxKKAEgTLiih1p75NHPCCgwlKWBvc9LFQ/edit#

Potential solution:
the Salesforce user would need the sync, not to trigger the creation of an "SF organization" when a *Fabrica account* is created from Salesforce, so to avoid potential duplication of organizations is salesforce.

**Comment by mfenner** [comment]
@kjgarza please propose a solution for how this should be implemented from a product engineering perspective. `Do not trigger the sync when an organization is created.` will not work. 

I removed the `bug` label, as this is not a bug.

**Comment by kjgarza** [comment]
@mfenner  let me rephrase that because maybe it is misunderstood. Basically we just need to avoid conflicts with the creation of accounts and organizations. 

Now, we do not need anymore that `the sync` creates organizations in salesforce. 

So basically, the Salesforce user would need the sync, not to trigger the creation of an "SF organization" when a *Fabrica account* is created from Salesforce, so to avoid potential duplication of organizations is salesforce.

**Comment by mfenner** [comment]
> Now, we do not need anymore that the sync creates organizations in salesforce.

I am not sure I understand this. A very important use case is creating and managing consortium organizations, which I assume with happen in Fabrica and not Salesforce. And I am not sure we want to make it impossible to create organizations (e.g. consortium or direct member) in Fabrica.

I don't understand the term `the sync`, as no syncing takes place, rather a one-way export.

**Comment by mjbuys** [comment]
As a workflow rule, only Consortium Organizations should be created in
Fabrica first. As mentioned in the team meeting, we should *not* create
production or test credentials in Fabrica (always in our CRM first,
thereafter updates happen in Fabrica as the source of truth. Is there a way
that the "create" trigger would only be for Consortium Organizations? This
would resolve the issue.

**Comment by kjgarza** [comment]
I think then maybe a better phrasing of the change is:

The salesforce user would need that the `member/provider export` would **ONLY** create new SalesForce organizations when a Fabrica account to be exported is member_type **consortium_organization**, so to avoid potential duplication of organizations created from Salesforce (from all other member_types).

**Comment by mfenner** [comment]
An alternative approach would be to have an optional "don't trigger salesforce update" flag on account creation. Which AppliconTech would include with all Lupo REST API call coming from Salesforce. I would prefer this approach.

**Comment by kjgarza** [comment]
does that flag exist?

**Comment by mfenner** [comment]
No, but I would implement it if we agree on the strategy.

**Comment by kjgarza** [comment]
I mean this is implementation, if you find this the best solution it would be fine. Just let me know the `attribute-value pair` to use and I'll pass it over to applikon-tech.

Would this approach (flag-attribute) require dev-work in `Fabrica` as well?

**Comment by mfenner** [comment]
@kjgarza I added a new optional parameter when creating/updating `provider`, `repository` or `contact` (using the JSONAPI format): 

```
"fromSalesforce": true
```

Then this parameter is added, no update of Salesforce is triggered. Keep in mind that Salesforce updates still happen in the once-a-day update of all records triggered by a cron job.

**Comment by kjgarza** [comment]
Thanks @mfenner .

FYI @Loveprogramming10 this is the change that we will need to implement, I will follow up by email

**Comment by mfenner** [comment]
@kjgarza I double-checked in `stage` and `production` for `providers` and `update`, and this feature works as expected. I didn't check `create`, and I didn't check `clients` and `contacts`.

Can you please do more checks using this feature?

**Comment by kjgarza** [comment]
well the salesforce integration only works on `create`  (its only to created Providers)

but I'll do another test anyway
